만들면서 배우는 클린 아키텍처/code/buckpal/src/test/kotlin/io/reflectoring/buckpal/common/AccountTestData.kt

```diff
@@ -0,0 +1,38 @@
+package io.reflectoring.buckpal.common
+
+import io.reflectoring.buckpal.account.domain.Account
+import io.reflectoring.buckpal.account.domain.ActivityWindow
+import io.reflectoring.buckpal.account.domain.Money
+
+object AccountTestData {
+
+    fun defaultAccount(): AccountBuilder {
+        return AccountBuilder()
+            .withAccountId(Account.AccountId(42))
+            .withBaselineBalance(Money.of(999))
+            .withActivityWindow(
+                ActivityWindow(
+                    ActivityTestData.defaultActivity().build(),
+                    ActivityTestData.defaultActivity().build(),
+                )
+            )
+    }
+
+    class AccountBuilder {
+        private var accountId: Account.AccountId? = null
+        private var baselineBalance: Money? = null
+        private var activityWindow: ActivityWindow? = null
+
+        fun withAccountId(accountId: Account.AccountId) = apply { this.accountId = accountId }
+        fun withBaselineBalance(baselineBalance: Money) = apply { this.baselineBalance = baselineBalance }
+        fun withActivityWindow(activityWindow: ActivityWindow) = apply { this.activityWindow = activityWindow }
+
+        fun build(): Account {
+            requireNotNull(accountId) { "AccountId must not be null" }
+            requireNotNull(baselineBalance) { "Baseline balance must not be null" }
+            requireNotNull(activityWindow) { "ActivityWindow must not be null" }
+
+            return Account.withId(accountId!!, baselineBalance!!, activityWindow!!)
+        }
```

Comment:
@coderabbitai

해당 연산자를 제거하면, IntelliJ에서 다음과 같은 에러가 발생한다.

> Smart cast to 'Account. AccountId' is impossible, because 'accountId' is a mutable property that could have been changed by this time